### ISSUE TITLE
280: Advanced UI for Collections

### ISSUE THREAD

**Issue by tngreene** [X-Plane/XPlane2Blender#506]
I decided to move part of #450 here:

# Advanced UI Spec
- [x] Exportable Collections Enabled/Disabled separated into two groups
- [x] Disabled Group Hidden Behind disclose arrow

Plus after alpha.3, people may have additional feedback

**Comment by tngreene** [comment]
Another idea: Only show top level collections. I think this might not be the winner, but, I think it is worth exploring. It would certainly be quick.

**Comment by danklaue** [comment]
I would LOVE that approach.  I think having hierarchical collections would help my workflow a lot, but I would need to turn off all the "child" collections from the exporter, as it would add up quickly down there.  (I'm thinking of making collections out of discrete conceptual "objects", such as control surfaces, which would be partially residing as collections within collections.... but the sub-collections would make absolutely no sense to export.)

**Comment by danklaue** [comment]
Here's a screenshot of what happens when you try making use of hierarchical scene collections:

![image](https://user-images.githubusercontent.com/359923/72570293-26700180-3882-11ea-867d-a92da7e10bcb.png)


By way of explanation: My convention is, E1, E2, E3, E4, etc. for exterior objects, associated with their respective textures.  In the screenshot, "_CS" signifies, it's a "control surfaces" group, and "_LG" signifies, it's a "Landing Gear" group.  These parts have multiple sub-parts associated with the textures E1, E2, etc. png.   I have separate .blend files for "_CS" and "_LG" collections, and that's why they're greyed out in the XP2B collection sets.  

Point is, I'd rather not have them there at all.  They clutter up the view, and I can't imagine them ever being necessary.  That is, if you want to export one of those as a separate object, you would simply remove it from the hierarchy.

I'd really appreciate the exporter only looking at the top directory of collections, to determine what should get exported.

**Comment by tngreene** [comment]
I like the "only show top collections in UI", but I know some people like using the top collection for grouping

```
euro_buildings (organizing)
  |_skyscraper_1 (exportable)
  |_skyscraper_2 (exportable)
usa_buildings (organizing)
  |_skyscraper_3 (exportable)
```

Under a "top collection = show in UI" scheme, this workflow would be hampered.

**Comment by tngreene** [comment]
New plan:

Separate those marked exportable collection to a section that is always visible, and below that, have a section hidden behind a disclose arrow. When checked or unchecked the OBJ features for that exportable collection move between sections.

- Tidy
- Easy to find and understand
- No special naming patterns
- No special top level patterns
- Meant to only be set once, and use the eyeball feature of #423 to control export, so you don't have to mess around with this often

**Comment by tngreene** [comment]
Closed with 934106c3c2c01
